Crash midgame, possible TOPAddons incompat

```log
[22:38:40] [Netty Local Server IO #3/ERROR] [FML]: There was a critical exception handling a packet on channel theoneprobe
io.netty.handler.codec.EncoderException: java.lang.NoSuchMethodError: 'void mcjty.theoneprobe.network.NetworkTools.writeString(io.netty.buffer.ByteBuf, java.lang.String)'
	at io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:107)
	at io.netty.handler.codec.MessageToMessageCodec.write(MessageToMessageCodec.java:130)
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:825)
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:746)
	at io.netty.channel.DefaultChannelPipeline.write(DefaultChannelPipeline.java:1031)
	at io.netty.channel.Channel.write(Channel.java:272)
	at io.netty.channel.embedded.EmbeddedChannel.write(EmbeddedChannel.java:775)
	at io.github.lxgaming.sledgehammer.util.NetworkChannelHelper.writeAndFlush(NetworkChannelHelper.java:73)
	at net.minecraftforge.fml.common.network.FMLEmbeddedChannel.writeAndFlush(FMLEmbeddedChannel.java:533)
	at net.minecraftforge.fml.common.network.simpleimpl.SimpleNetworkWrapper.sendTo(SimpleNetworkWrapper.java:250)
	at mcjty.theoneprobe.network.PacketGetInfo$Handler.handle(PacketGetInfo.java:118)
	at mcjty.theoneprobe.network.PacketGetInfo$Handler.lambda$onMessage$0(PacketGetInfo.java:109)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:545)
	at java.util.concurrent.FutureTask.run(FutureTask.java:328)
	at net.minecraft.util.Util.runTask(SourceFile:529)
	at net.minecraft.server.MinecraftServer.updateTimeLightAndEntities(MinecraftServer.java:723)
	at net.minecraft.server.MinecraftServer.tick(MinecraftServer.java:668)
	at net.minecraft.server.integrated.IntegratedServer.tick(IntegratedServer.java:185)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526)
	at java.lang.Thread.run(Thread.java:1447)
Caused by: java.lang.NoSuchMethodError: 'void mcjty.theoneprobe.network.NetworkTools.writeString(io.netty.buffer.ByteBuf, java.lang.String)'
	at io.github.drmanganese.topaddons.elements.ElementTankGauge.toBytes(ElementTankGauge.java:83)
	at mcjty.theoneprobe.apiimpl.ProbeInfo.writeElements(ProbeInfo.java:45)
	at mcjty.theoneprobe.apiimpl.elements.AbstractElementPanel.toBytes(AbstractElementPanel.java:58)
	at mcjty.theoneprobe.network.PacketReturnInfo.toBytes(PacketReturnInfo.java:36)
	at net.minecraftforge.fml.common.network.simpleimpl.SimpleIndexedCodec.encodeInto(SimpleIndexedCodec.java:30)
	at net.minecraftforge.fml.common.network.simpleimpl.SimpleIndexedCodec.encodeInto(SimpleIndexedCodec.java:26)
	at net.minecraftforge.fml.common.network.FMLIndexedMessageToMessageCodec.encode(FMLIndexedMessageToMessageCodec.java:81)
	at io.netty.handler.codec.MessageToMessageCodec$2.encode(MessageToMessageCodec.java:85)
	at io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:90)
	... 19 more
```

--- a/src/main/java/mcjty/theoneprobe/network/NetworkTools.java
+++ b/src/main/java/mcjty/theoneprobe/network/NetworkTools.java
@@ -83,9 +83,15 @@ public class NetworkTools {
         }
     }
 
+    /**
+     * Writes a UTF-8 encoded String to the given ByteBuf. Can handle null strings.
+     *
+     * @param buf The ByteBuf to write to.
+     * @param str The String to write.
+     */
     public static void writeStringCompact(ByteBuf buf, String str) {
         if (str == null) {
-            writeVarInt(buf, 0); // length = 0 means empty/null
+            writeVarInt(buf, -1); // -1 = null
             return;
         }
 
@@ -94,14 +100,68 @@ public class NetworkTools {
         buf.writeBytes(bytes);
     }
 
+    /**
+     * Reads a UTF-8 encoded String from the given ByteBuf.
+     *
+     * @param buf The ByteBuf to read from.
+     * @return The String read from the buffer, or null if the length was -1, or an empty string if the length was 0.
+     */
     public static String readStringCompact(ByteBuf buf) {
         int length = readVarInt(buf);
+        if (length == -1) return null;
         if (length == 0) return "";
         byte[] bytes = new byte[length];
         buf.readBytes(bytes);
         return new String(bytes, java.nio.charset.StandardCharsets.UTF_8);
     }
 
+    /**
+     * Writes a UTF-8 encoded String to the given ByteBuf. Can handle null strings.<br>
+     * For compat only. Use {@link NetworkTools#writeStringCompact} instead.
+     *
+     * @param buf The ByteBuf to write to.
+     * @param str The String to write.
+     */
+    @Deprecated
+    public static void writeString(ByteBuf buf, String str) {
+        writeStringCompact(buf, str);
+    }
+
+    /**
+     * Writes a UTF-8 encoded String to the given ByteBuf. Can handle null strings.<br>
+     * For compat only. Use {@link NetworkTools#writeStringCompact} instead.
+     *
+     * @param buf The ByteBuf to write to.
+     * @param str The String to write.
+     */
+    @Deprecated
+    public static void writeStringUTF8(ByteBuf buf, String str) {
+        writeStringCompact(buf, str);
+    }
+
+    /**
+     * Reads a UTF-8 encoded String from the given ByteBuf.<br>
+     * For compat only. Use {@link NetworkTools#readStringCompact} instead.
+     *
+     * @param buf The ByteBuf to read from.
+     * @return The String read from the buffer, or null if the length was -1, or an empty string if the length was 0.
+     */
+    @Deprecated
+    public static String readString(ByteBuf buf) {
+        return readStringCompact(buf);
+    }
+
+    /**
+     * Reads a UTF-8 encoded String from the given ByteBuf.<br>
+     * For compat only. Use {@link NetworkTools#readStringCompact} instead.
+     *
+     * @param buf The ByteBuf to read from.
+     * @return The String read from the buffer, or null if the length was -1, or an empty string if the length was 0.
+     */
+    @Deprecated
+    public static String readStringUTF8(ByteBuf buf) {
+        return readStringCompact(buf);
+    }
 
     /**
      * Reads a BlockPos from the given ByteBuf.

